rotkehlchen/api/rest.py

```diff
@@ -3910,9 +3912,13 @@ def _query_kraken_staking_events(
             Location.KRAKEN,
         )
         message = ''
-        if exchanges_list is None:
-            events = []
-        else:
+        events: List[StakingEvent] = []
+        entries_found = 0
+        with_limit, entries_limit = False, -1
+        if self.rotkehlchen.premium is None:
+            with_limit, entries_limit = True, FREE_HISTORY_EVENTS
+
+        if exchanges_list is not None:
```

Comment:
Why have this here like that? It forces you to put the main logic more and more indented.

If exchanges_list is None then you don't need to do anything right?

So have it at the very very top with

```python
if exchanges_list is None:
    return {'whatever_empty_response_looks_like'}
```